daemon/api.go

```diff
@@ -789,9 +790,11 @@ func withEnsureUbuntuCore(st *state.State, targetSnap string, userID int, instal
 }
 
 var errModeConflict = errors.New("cannot use devmode and jailmode flags together")
+var errClassicJailmodeConflict = errors.New("cannot use classic and jailmode flags together")
```

Comment:
Reading the code, it's starting to feel like that's not a good idea. For example, the design of the client.Snap interface is being made considering that you cannot alter the snap's opinion of whether it is installed as a classic snap or not, and thus our checks are being made via the snap's confinement request itself instead of the runtime decision of whether we allowed it to be classic or not. This seems unwise.. people will also create code like this, and then it will break when we finalize the design.

Let's please go ahead and allow jailmode and classic together, with the straightforward meaning: simply put the classic snap under normal confinement rules. This will force us to design the APIs correctly.